### PR TITLE
New package: diod-1.0.24

### PR DESCRIPTION
**Pull request by pulux**

### CODE REVIEW

File: srcpkgs/diod/template [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Template file for 'diod'
+pkgname=diod
+version=1.0.24
+revision=1
+build_style=gnu-configure
+hostmakedepends="automake autoconf"
+makedepends="libcap-devel ncurses-devel attr-devel popt-devel gperftools-devel lua51-devel munge-devel"
```
Comment (xtraeme) [inline srcpkgs/diod/template]: Any reason to use `lua51` and not `lua (5.3)` ?

File: srcpkgs/diod/template [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Template file for 'diod'
+pkgname=diod
+version=1.0.24
+revision=1
+build_style=gnu-configure
+hostmakedepends="automake autoconf"
```
Comment (xtraeme) [inline srcpkgs/diod/template]: `autoconf` is a dependency of `automake`, it's redundant.

File: srcpkgs/diod/template (hunk 1 above)
Comment (pulux) [inline srcpkgs/diod/template]: the pkg don't build here not with lua:

```
../libdiod/libdiod.a(diod_conf.o): In function `_lua_getglobal_string':
diod_conf.c:(.text+0x31d): undefined reference to `lua_getglobal'
...
diod_conf.c:(.text+0x143a): undefined reference to `lua_tolstring'
../libdiod/libdiod.a(diod_conf.o): In function `_lua_get_expattr':
diod_conf.c:(.text+0x4d5): undefined reference to `lua_settop'
collect2: error: ld returned 1 exit status
Makefile:358: recipe for target 'diod' failed
make[1]: *** [diod] Error 1
make[1]: Leaving directory '/builddir/diod-1.0.24/diod'
Makefile:332: recipe for target 'all-recursive' failed
make: *** [all-recursive] Error 1
=> ERROR: diod-1.0.24_1: failed to run do_build() at line 13.
```

Comment (pulux) [inline srcpkgs/diod/template]: the pkg don't build here with lua-devel:
```/usr/include/features.h:148:3: warning: #warning "_BSD_SOURCE and _SVID_SOURCE are deprecated, use _DEFAULT_SOURCE" [-Wcpp]
 # warning "_BSD_SOURCE and _SVID_SOURCE are deprecated, use _DEFAULT_SOURCE"
   ^
  CC     fid.o
  CC     xattr.o
  CCLD   diod
../libdiod/libdiod.a(diod_conf.o): In function `_lua_getglobal_string':
diod_conf.c:(.text+0x31d): undefined reference to `lua_getglobal'
...
diod_conf.c:(.text+0x143a): undefined reference to `lua_tolstring'
../libdiod/libdiod.a(diod_conf.o): In function `_lua_get_expattr':
diod_conf.c:(.text+0x4d5): undefined reference to `lua_settop'
collect2: error: ld returned 1 exit status
Makefile:358: recipe for target 'diod' failed
make[1]: *** [diod] Error 1
make[1]: Leaving directory '/builddir/diod-1.0.24/diod'
Makefile:332: recipe for target 'all-recursive' failed
make: *** [all-recursive] Error 1
=> ERROR: diod-1.0.24_1: failed to run do_build() at line 13.
```

File: srcpkgs/diod/template [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# Template file for 'diod'
+pkgname=diod
+version=1.0.24
+revision=1
+build_style=gnu-configure
+hostmakedepends="autoconf"
```
Comment (xtraeme) [inline srcpkgs/diod/template]: `conf_files=/etc/diod.conf`

**Pull request merged**